agent/03_agent_deployment.sh

```diff
@@ -107,17 +150,61 @@ stringData:
   .dockerconfigjson: '${pull_secret}'
 
 EOF
+
+    num_ips=${#FLEETING_NODES_IPS[@]}
+
+    # Create a yaml for each host in nmstateconfig.yaml
+    for (( i=0; i<$num_ips; i++ ))
+    do
+        cat >> "${FLEETING_MANIFESTS_PATH}/nmstateconfig.yaml" << EOF
+apiVersion: agent-install.openshift.io/v1beta1
+kind: NMStateConfig
+metadata:
+  name: ${FLEETING_NODES_HOSTNAMES[i]}
+  namespace: openshift-machine-api
+  labels:
+    cluster0-nmstate-label-name: cluster0-nmstate-label-value
+spec:
+  config:
+    interfaces:
+      - name: eth0
+        type: ethernet
+        state: up
+        mac-address: ${FLEETING_NODES_MACS[i]}
+        ipv4:
+          enabled: true
+          address:
+            - ip: ${FLEETING_NODES_IPS[i]}
+              prefix-length: ${CLUSTER_HOST_PREFIX}
+          dhcp: false
+    dns-resolver:
+      config:
+        server:
+          - ${PROVISIONING_HOST_EXTERNAL_IP}
+    routes:
+      config:
+        - destination: 0.0.0.0/0
+          next-hop-address: ${PROVISIONING_HOST_EXTERNAL_IP}
+          next-hop-interface: eth0
+          table-id: 254
+  interfaces:
+    - name: "eth0"
+      macAddress: ${FLEETING_NODES_MACS[i]}
+---
+EOF
+    done
+
     set -x
 }
 
 function generate_fleeting_iso() {
     export REPO_PATH=${WORKING_DIR}
+
     sync_repo_and_patch fleeting https://github.com/openshift-agent-team/fleeting ${FLEETING_PR}
 
-    generate_fleeting_manifests ${FLEETING_PATH}
-    
+    generate_fleeting_manifests ${FLEETING_PATH} ${FLEETING_NODES_IPS}
```

Comment:
nit: this could be simplified, but we could do it into another pr

Reply:
Good catch, updated.